### ISSUE TITLE
JSONObject["stack"] is not a JSONArray Error while fetching jobs with error in sjs client

### ISSUE THREAD

**Issue by knvs97** [bluebreezecf/SparkJobServerClient#28]
Error Job response
[{
  "duration": "-19143.958 secs",
  "classPath": "spark.jobserver.TestJob",
  "startTime": "2018-07-13T02:47:20.540Z",
  "context": "context1",
  "result": {
    "message": "Unexpected termination of context context1",
    "errorClass": "",
    "stack": ""
  },
  "status": "ERROR",
  "jobId": "483a0205-a72b-452b-95c1-68db1ff34230"
}]
The stack is empty but trying to initialize into JSONArray is throwing error.

**Comment by bluebreezecf** [comment]
@knvs97  Thanks for your detailed information.

**Comment by roybirger** [comment]
This still looks like an issue: (JSONArray initialization)

```
net.sf.json.JSONException: JSONObject["stack"] is not a JSONArray.
	at net.sf.json.JSONObject.getJSONArray(JSONObject.java:2038)
	at com.bluebreezecf.tools.sparkjobserver.api.SparkJobServerClientImpl.setErrorDetails(SparkJobServerClientImpl.java:683)
	at com.bluebreezecf.tools.sparkjobserver.api.SparkJobServerClientImpl.getJobs(SparkJobServerClientImpl.java:334)
	...
```

**Comment by knvs97** [comment]
> This still looks like an issue: (JSONArray initialization)
> 
> ```
> net.sf.json.JSONException: JSONObject["stack"] is not a JSONArray.
> 	at net.sf.json.JSONObject.getJSONArray(JSONObject.java:2038)
> 	at com.bluebreezecf.tools.sparkjobserver.api.SparkJobServerClientImpl.setErrorDetails(SparkJobServerClientImpl.java:683)
> 	at com.bluebreezecf.tools.sparkjobserver.api.SparkJobServerClientImpl.getJobs(SparkJobServerClientImpl.java:334)
> 	...
> ```

@roybirger can you please elaborate?

**Comment by roybirger** [comment]
this line (.../sparkjobserver/api/SparkJobServerClientImpl.java:661)

`JSONArray stackJsonArray = resultJson.getJSONArray(SparkJobInfo.INFO_KEY_RESULT_STACK);`

throws an exception when stackJsonArray @ SparkJobInfo.INFO_KEY_RESULT_STACK is a string and not a json array.

**Comment by knvs97** [comment]
Yes, the problem will still exist if the stack is a string @roybirger.  I did try out with simple test jars. So did not try out for a case with data inside stack. Thanks for notifying the mistake @roybirger .

**Comment by bluebreezecf** [comment]
@roybirger  @knvs97  The discutions above seem meaningful

**Comment by knvs97** [comment]
This is resolved with the Merge Request #33 by @roybirger .
